app/Questionnaire.js

```diff
@@ -1112,6 +1165,38 @@ $.extend( Questionnaire.prototype, {
 	},
 
 	/**
+	 * Updates the navigation after logging an answer.
+	 *
+	 * @param {string} page
+         * @param {string} toPage
+	 */
+        _updateNavigation: function( page, toPage ) {
+                //christoph.fischer
+                var updatedNavigationEntry = {
+                            page: page,
+                            loggedAnswers: $.extend( {}, this._loggedAnswers )
+                    },
+                    nextNavigationEntry = {
+                            page: toPage,
+                            loggedAnswers: ({}, {})
+                    };
+
+
+                this._navigationCache.splice(  this._navigationState );
+
+                if( this._navigationCache.length === 0 ) {
+                    this._navigationCache.push( updatedNavigationEntry );
+                } else {
+                    this._navigationCache[this._navigationState] = updatedNavigationEntry;
+                }
+
+                this._navigationCache.push( nextNavigationEntry );
+
+                this._navigationState++;                        
```

Comment:
I wonder if this code would work if all instances of `_navigationState` are replaced with `_navigationCache.length`?